lib/romeo/auth.ex

```diff
@@ -44,8 +44,16 @@ defmodule Romeo.Auth do
 
 
   defp do_authenticate(mechanism, conn) do
-    Logger.info fn -> "Authenticating with #{mechanism}" end
-    authenticate_with(mechanism, conn)
+    {:ok, conn} =
+      case mechanism do
+        {name, mechanism_fn} ->
```

Comment:
Another option that might allow for more flexibility would be to pass the mechanism as well. So `authenticate/2`.

That way in the case of your example, you could use:

```elixir
preferred_auth_mechanisms: [{"X-OAUTH2", __MODULE__}, {"CUSTOM", __MODULE__}]
```

```elixir
def authenticate("X-OAUTH2", conn), do: ...
def authenticate("CUSTOM", conn), do: ...
```